honeybee_radiance/cli/multiphase.py

```diff
@@ -0,0 +1,113 @@
+"""honeybee radiance ray-tracing command commands."""
+import click
+import sys
+import logging
+import os
+import traceback
+
+
+from honeybee_radiance.config import folders
+from honeybee_radiance_command.rfluxmtx import RfluxmtxOptions, Rfluxmtx
+
+
+_logger = logging.getLogger(__name__)
+
+
+@click.group(help="Commands to run multi-phase operations in Radiance.")
+def multi_phase():
+    pass
+
+
+@multi_phase.command("view-matrix")
+@click.argument(
+    "receiver-file", type=click.Path(exists=True, file_okay=True, resolve_path=True)
+)
+@click.argument(
+    "octree", type=click.Path(exists=True, file_okay=True, resolve_path=True)
+)
+@click.argument(
+    "sensor-grid", type=click.Path(exists=True, file_okay=True, resolve_path=True)
+)
+@click.option(
+    "--sensor-count",
+    type=click.INT,
+    show_default=True,
+    help="Number of sensors in sensor grid file. Number of sensors will be parsed form"
+    " the sensor file if not provided.",
+)
+@click.option("--rad-params", show_default=True, help="Radiance parameters.")
+@click.option(
+    "--rad-params-locked",
+    show_default=True,
+    help="Protected Radiance parameters. "
+    "These values will overwrite user input rad parameters.",
+)
+@click.option(
+    "--output",
+    "-o",
+    show_default=True,
+    help="Path to output file. If a relative path"
+    " is provided it should be relative to project folder.",
+)
+@click.option(
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="A flag to show the command without running it.",
+)
+def view_matrix_command(
+    receiver_file,
+    octree,
+    sensor_grid,
+    sensor_count,
+    rad_params,
+    rad_params_locked,
+    output,
+    dry_run,
+):
+    """Calculate view matrix for a receiver file.
+
+    \b
+    Args:
+        receiver_file: Path to receiver file.
+        octree: Path to octree file.
+        sensor_grid: Path to sensor grid file.
+    """
+
+    try:
+        options = RfluxmtxOptions()
+        # parse input radiance parameters
+        if rad_params:
+            options.update_from_string(rad_params.strip())
+        # overwrite input values with protected ones
+        if rad_params_locked:
+            options.update_from_string(rad_params_locked.strip())
+
+        if not sensor_count:
+            raise ValueError("for the time-being sensor count must be provided.")
```

Comment:
I'm inclined to think that we shouldn't merge this until we address this. If the sensor-count is not provided, we just need just a few lines of code to count the number of lines in the file, right?

I think we should probably also add something to the input description to say why you would supply this sensor-count. It's just to make the calculation faster, correct?

Reply:
Thanks, @chriswmackey!

@mikkelkp, is it something that you can address? It will be a helper function. We already have such functions that read sensor grids from a file.

If that's the case my suggestion is to merge this one and then update this command and a couple of other commands which have the same error message.

Reply:
Sure, @mostaphaRoudsari. I assume you think of [from_file](https://github.com/ladybug-tools/honeybee-radiance/blob/8804c79a991c3160484b510198de14a59bda028e/honeybee_radiance/sensorgrid.py#L182) and then get the sensor count by the count property.

I can add it to the other commands as well.

Reply:
Hi @mikkelkp! Yes. But with the difference that you don't need to create the sensor grids which is slower. You just need to go through the file and count the number of lines while making sure you don't count the comment lines if any.